reyna/src/com/b2msolutions/reyna/system/Preferences.java

```diff
@@ -196,4 +199,25 @@ private String getString(String key, String defaultValue) {
         SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
         return sp.getString(key, defaultValue);
     }
+
+    public long getNonRecurringWwanBlackoutStartTime() {
+        return this.getLong(WWAN_BLACKOUT_START, -1);
```

Comment:
I was just discussing this, at the moment the save function saves both so for example if you wanted blackout from -1 to 1 you can and similarly you can blackout -2 to -1. Just if both values are not set you will get both being -1 and therefore return OK. But I might try to make this more explicit.